### PR TITLE
Add libcouchbase 2.5.7

### PR DESCRIPTION
**Pull request by avsej**

### CODE REVIEW

File: components/library/libcouchbase/patches/02-multiarch-pkg_config.patch [hunk 1]
```diff
@@ -0,0 +1,12 @@
+--- libcouchbase-2.5.7/packaging/libcouchbase.pc.in.orig	2016-01-12 00:44:09.000000000 +0300
```
Comment (pyhalov) [inline components/library/libcouchbase/patches/02-multiarch-pkg_config.patch]: Can we use more elegant patch here, for example, refer CMAKE_INSTALL_BINDIR ?

File: components/library/libcouchbase/Makefile [hunk 2]
```diff
@@ -0,0 +1,64 @@
+#
```
Comment (pyhalov) [inline components/library/libcouchbase/Makefile]: use illumos header

File: components/library/libcouchbase/resolve.deps [hunk 3]
```diff
@@ -0,0 +1,3 @@
+library/libev
```
Comment (pyhalov) [inline components/library/libcouchbase/resolve.deps]: oi-userland doesn't use resolve.deps

File: components/library/libcouchbase/Makefile [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+CMAKE_OPTIONS+=	-DLCB_NO_TESTS=1
+
+COMPONENT_POST_BUILD_ACTION += \
+	case $(BITS) in \
+	32) $(GSED) -i -e "s:/MACH64::g"         $(BUILD_DIR_32)/packaging/libcouchbase.pc;; \
+	64) $(GSED) -i -e "s:MACH64:$(MACH64):g" $(BUILD_DIR_64)/packaging/libcouchbase.pc;; \
+	esac;
+
+# common targets
+build:		$(BUILD_32_and_64)
```
Comment (pyhalov) [inline components/library/libcouchbase/Makefile:40]: Have you compared 32-bit and 64-bit headers? Are they the same?

Comment (avsej) [inline components/library/libcouchbase/Makefile:40]: yes, they are identical

**Comment by avsej** [comment]
@pyhalov thanks for pointing, patch has been updated

**Pull request merged**